### ISSUE TITLE
Return empty response instead of 404 for no trades/market holidays - Historic Quotes ( NBBO )

### ISSUE THREAD

**Issue by sharok** [polygon-io/issues#23]
**Is your feature request related to a problem? Please describe.**
Sometimes request to `/v2/ticks/stocks/nbbo/{ticker}/{date}` returns 404 and the following response:

> {
>    "error":"Unknown database error ( Error Code: 001 )",
>    "errorcode":"001",
>    "status":"ERROR",
>    "success":false
> }

After some testing, I have figured out that this response happens when the market was closed( holiday).

The current behavior says that something went wrong and some libraries throw exceptions because status is 404.

**Describe the solution you'd like**
If there is no data for the specific day, then the response should be 200 and contains an empty `results` field.

**Comment by jrbell19** [comment]
This is actually something that we are planning to make more clear in the responses we send out.

**Comment by qrpike** [comment]
@jbonzo thoughts? This DB code is thrown when we do not have any data for that date at all, for any ticker ( weekends / holidays ). 

If there is a DB table for that date, then we would return an empty array ( with 200 ) if there is no data.

**Comment by jbonzo** [comment]
Yeah I think a good way to frame it what does it mean for `/v2/ticks/stocks/nbbo/AAPL/2020-05-26` to `404`. I think you could either interpret that as the resource AAPL stocks on 2020-05-26 are not found *or* you could interpret it as the resource AAPL stocks on any day cannot be found, since there are two resource identifiers in the endpoint.

I think for this endpoints, and others like it, the `date` parameter isn't regarded as a resource identifier since the resource is identified by the `ticker` and each date is a different instance in a timeseries. I think our endpoints should follow the following logic:

For a given ticker `t` where `t` does exist, then return a `404` since there is no resource for that request URI. For any date `d` for a ticker `t` that does exist, return a `200` with an empty set as the result body.

Tl;DR
* If ticker exists always 200
* If ticker doesn't exist then 404

**Comment by spazmodius** [comment]
Empty results are the right response for _valid tickers_ during _time periods in which no quotes occurred_. This would include weekends, holidays, overnight, maybe even trading halts.

However, for genuinely missing data from Polygon's database, a non-200 response is more correct. It is the equivalent of throwing an exception, because the normal return value cannot hold the exceptional result.

**Comment by KshitizSharmaV** [comment]
Data not available for 15th June for any of the tickers

https://api.polygon.io/v2/ticks/stocks/nbbo/AAPL/2020-06-15?limit=10&apiKey=XXXXXXXXX

{"error":"Unknown database error 
( Error Code: 001 _)",
"errorcode":"001",
"status":"ERROR",
"success":false}

Please fix this issue, API keeps failing again & again for NBBO data. This is a well-known issue, historic quotes are important for any kind of analysis to estimate the cost of a trade.
